examples/passport-login/src/authentication-strategies/provider.ts

```diff
@@ -5,31 +5,62 @@
 
 import {extensions, Getter, Provider, inject} from '@loopback/core';
 import {Oauth2AuthStrategy} from './oauth2-strategy';
-import {Request, RestBindings, HttpErrors} from '@loopback/rest';
+import {Request, RestBindings} from '@loopback/rest';
+import {
+  AuthenticationStrategy,
+  AuthenticationBindings,
+  AuthenticationMetadata,
+} from '@loopback/authentication';
 
 export namespace PassportAuthenticationBindings {
   export const OAUTH2_STRATEGY = 'passport.authentication.oauth2.strategy';
+  export const PASSPORT_STRATEGY = 'passport.authentication.strategy';
 }
 
 /**
- * provider for passport strategies
+ * provider for oauth2 strategies
  */
-export class PassportStrategyProvider implements Provider<Oauth2AuthStrategy> {
+export class Oauth2StrategyProvider
+  implements Provider<Oauth2AuthStrategy | undefined> {
   constructor(
     @extensions(PassportAuthenticationBindings.OAUTH2_STRATEGY)
     private getStrategies: Getter<Oauth2AuthStrategy[]>,
     @inject(RestBindings.Http.REQUEST)
     private request: Request,
   ) {}
 
-  async value() {
+  async value(): Promise<Oauth2AuthStrategy | undefined> {
     const providerName = this.request.route.provider;
     const strategies: Oauth2AuthStrategy[] = await this.getStrategies();
     const strategy = strategies.find(
       (s: Oauth2AuthStrategy) => s.provider === providerName,
     );
+    return strategy;
+  }
+}
+
+/**
+ * provider for other passport strategies
+ */
+export class PassportStrategyProvider
+  implements Provider<AuthenticationStrategy | undefined> {
+  constructor(
+    @extensions(PassportAuthenticationBindings.PASSPORT_STRATEGY)
+    private getStrategies: Getter<AuthenticationStrategy[]>,
+    @inject(AuthenticationBindings.METADATA)
+    protected metadata?: AuthenticationMetadata,
+  ) {}
 
-    if (!strategy) throw new HttpErrors.Unauthorized('unknown provider');
+  async value(): Promise<AuthenticationStrategy | undefined> {
+    if (!this.metadata) return;
+    const name = this.metadata.strategy;
+    const strategy = await this.findAuthenticationStrategy(name);
     return strategy;
   }
+
+  async findAuthenticationStrategy(name: string) {
```

Comment:
i am a little confused here. We already have an authentication strategy provider. https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/providers/auth-strategy.provider.ts#L41 .

Reply:
agreed, removing this now.